src/util-logopenfile.c

```diff
@@ -83,9 +129,21 @@ static int SCLogFileWrite(const char *buffer, int buffer_len, LogFileCtx *log_ct
 
     int ret = 0;
 
+    if (log_ctx->fp == NULL && log_ctx->is_sock)
+        ret = SCLogUnixSocketReconnect(log_ctx);
```

Comment:
SCLogUnixSocketReconnect returns zero iff log_ctx->fp is NULL. Hence, it returns the same value as before: zero on failure; otherwise the result of fwrite.